### PR TITLE
Feature/fingerbank via connector

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Adds support for the Fingerbank Collector on a pfconnector-remote instance

# Impacts
Fingerbank in PacketFence

# Code / PR Dependencies
Needs the pfconnector-remote debian packaging to work #7232 

# NEW Package(s) required
fingerbank-collector-1.4.1 and fingerbank-collector-remote-1.4.1 (both in devel repos)

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [X] Document the feature
- [N/A] Add unit tests
- [X] Add acceptance tests (TestLink)

# NEWS file entries
(REQUIRED, but may be optional...)
## New Features
* Added support for the Fingerbank Collector on a PacketFence connector

### CODE REVIEW

**Comment by julsemaan** [comment]
# TODO

- [ ] High level schema for how it works
- [x] Add fingerbank-collector-remote as a dependency to packetfence-pfconnector-remote

**Comment by jrouzierinverse** [comment]
Looks good so far. What is the best way to test?

File: addons/pfconnector/configure.sh [hunk 1]
```diff
@@ -41,3 +41,4 @@ if ! prompt "Should the pfconnector server TLS certificate be validated?"; then
 fi
 
 echo "FETCH_REMOTES_VIA_API=true" >> /etc/pfconnector-client.env
+echo "PFCONNECTOR_REMOTE=true" >> /etc/pfconnector-client.env
```
Comment (nqb) [inline addons/pfconnector/configure.sh]: We will need to rebase this PR on `devel` once #7232 is merged because this path will not be valid.

**Comment by julsemaan** [comment]
> Looks good so far. What is the best way to test?

It's pretty manual now since the packaging for the pfconnector-remote is not completed yet but here are the general guidelines

- Install the pfconnector-remote from this branch
- Install fingerbank-collector-remote
- Connect your pfconnector-remote to PF
- Test some stuff related to Fingerbank for a node behind the pfconnector-remote

**Review by nqb — APPROVED** [review]

**Comment by julsemaan** [comment]
@fdurand and @jrouzierinverse, can you provide a review of the code?

Thanks!

File: lib/pf/iptables.pm [hunk 2]
```diff
@@ -154,6 +154,18 @@ sub iptables_generate {
     }
     $tags{'netdata'} .= "-A input-management-if --protocol tcp --match tcp --dport 19999 --jump DROP\n";
 
+    # The dynamic range used to access the fingerbank collector that are connected via a remote connector
+    $tags{'pfconnector'} = "";
```
Comment (fdurand) [inline lib/pf/iptables.pm:158]: i don´t see pfconnector tag in iptables.conf.example

File: go/chisel/server/server_handler.go [hunk 3]
```diff
@@ -253,22 +278,165 @@ func (s *Server) handleDynReverse(w http.ResponseWriter, req *http.Request) {
...
+	connectorId := req.URL.Query().Get("connector-id")
+	if connectorId == "" {
+		w.WriteHeader(http.StatusBadRequest)
+		json.NewEncoder(w).Encode(unifiedapiclient.ErrorReply{Status: http.StatusNotFound, Message: "Missing connector-id query parameter"})
+		return
+	}
+
+	if o, ok := activeTunnels.Load(connectorId); ok {
+		tun := o.(*tunnel.Tunnel)
+		index := s.computeConnectorIndex(connectorId)
```
Comment (jrouzierinverse) [inline go/chisel/server/server_handler.go:294]: It looks like this index only increases.

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

File: go/chisel/server/server_handler.go (hunk 3 above)
Comment (julsemaan) [reply go/chisel/server/server_handler.go:294]: It does, I'm assuming that we're connecting less than 256 connectors to a single server which seems like a fair limit for the moment

Comment (jrouzierinverse) [reply go/chisel/server/server_handler.go:294]: Can connectors be removed?

Comment (julsemaan) [reply go/chisel/server/server_handler.go:294]: They can but it's pretty unlikely a lot of them get removed. Worst case, the service can be restarted to clear the counter

File: lib/pf/iptables.pm (hunk 2 above)
Comment (julsemaan) [reply lib/pf/iptables.pm:158]: Good catch, it's fixed

**Review by fdurand — APPROVED** [review]

**Pull request merged**